### DISCUSSION TITLE
[Bug]: high power usage on Rak 4631/19003 & GNSS 12500

### DISCUSSION THREAD

**Discussion by LanWolf** [meshtastic/firmware#4102] [Troubleshooting]
### Category

Other

### Hardware

Rak4631

### Firmware Version

2.3.12

### Description

On 2.3.9 and 2.3.12 I experienced way higher power usage than with 2.2.23.

Usage on 2.2.23 I see 15ma typical with gps off,
On 2.3.12 I see arround 28ma with gps off.

This almost halves battery runtime on my portable node from almost  a week to only 3 days with moderate usage and no gps.

With gps on there is not much of a difference, 65ma for 2.2.23 and 68ma for 2.3.12

**Comment by thebentern** [comment]
@LanWolf which WisBlock GPS model are you running?

**Reply by LanWolf** [reply]
Sorry, only posted that in the title, it is the RAK 12500
BaseBoard is the 19003.

**Reply by thebentern** [reply]
> Sorry, only posted that in the title, it is the RAK 12500 BaseBoard is the 19003.

Oh I completely missed that, sorry 😅

**Comment by GPSFan** [comment]
Is the ~.6ma the difference between on and off?
What did you use to measure those numbers?
How are you turning the GPS on and off?

**Reply by LanWolf** [reply]
Sorry, I did write the numbers down wrong, it should be 65ma and 68ma with gps ON. 

The 3ma difference is between the GPS ON state on v 2.2.23 being 65ma and 2.3.12 with 2.3.12 being 68ma.

I first tested with a usb power meter, as the result from the behavoiur I saw (going back form a week to arround 3 day) did not ask for very precise measurement.

When seeing just the small difference in the gps on I retested with a peak tech 2025 factory calibrated digital multimeter to be sure there was no measuring error introduced. Results were the same range then too, after letting the device rest and stable out for a few minutes after power on.  Arrond 3ma with gps on, and 12ma difference with gps off 

I turn the gps on and off in the app, but also tried the tripple button click.
On both it says gps disabled on the screen of the node.

**Comment by GPSFan** [comment]
Thanks for the clarification.
What was the state of the GPS when you did your measurements, did it have a Lock? did it have a good view of the sky, Indoors or outdoors? how long had the setup been running?
 The DMM will give you good AVERAGE readings but especially with LORA and GPS there are lots of very high peak currents that can throw off an averaging reading meter.
With that said, i have a setup similar to what you described, RAK 4361, 12500 but a different base board. I have a Nordic Power Profiler 2 which gives a good view of high speed transients as well as recording capability. I will be doing some tests to try and replicate your results. There is a lot of on-going work in the power management of GPS area, and 2.3.12 had some issues that need addressing.

**Reply by LanWolf** [reply]
I always wait a few minutes for things to settle down when measuring, but my issue is big not with gps On, but with gps OFF.  That 3ma difference is without a lock.
The peaktech software saves  the readings over longer time, so I can eliminate peaks from sending Lora or Bt and occasional lows to get a pretty good reading.

**Comment by GPSFan** [comment]
Ok, so your problem seems to be in the disabling, The RAK is able to switch the 3V3_S to the GPS on and off via a GPIO (IO2), But it also powers other peripherals, I'll have to review the code to see how the GPS Disabled is handled. There was discussion at one point about whether using this was a good idea, and just to rely on disabling the GPS in software.

**Reply by thebentern** [reply]
I believe the 3V3_S pin toggling was dropped from the firmware because of the problematic nature of interfering with multiple wisblock modules, and now we rely on soft power saving commands for the GPS's low power state.

**Reply by LanWolf** [reply]
I can imagine that other wisblock modules like environmental sensor or rtc would need the 3v3 but imho that would not be used together ? For a portable node I would need gps, and for a router an rtc and environment sensor would make sense. 

Would be using the 3v3_S pin be  selectable ? I see the option to enter the pin number in the software for gps enable/disable.  If that would be left empty that would only soft en/disable, and with it populated with the right pin it would also disable the 3v3 ?

**Reply by thebentern** [reply]
It is a hardware variant macro. I'm not sure that setting gps en gpio on config is going to produce predictable results.

**Comment by GPSFan** [comment]
That is my recollection as well. 
In the discussion in:
https://github.com/meshtastic/firmware/pull/4062
I discovered that the PMREQ command was not being completed when the ESP32 was sent into Deep Sleep mode. There may be a similar issue with the nrf52.
This is an on going investigation which will be addressed asap.

**Comment by GPSFan** [comment]
You should try the latest 2.3.14 bleeding edge, there were some issues with 2.3.12 WRT GPS updating. I'd like to see your power numbers for this version. Because of the RAK 12500 design, the antenna will always be powered when the GPS has VCC. Not something we can do anything about. A hardware re-design of the RAK12500 would allow software sleep of the GPS to be truly low power.

**Comment by LanWolf** [comment]
Sorry for late response, having some health issues isues that prevented me from being that active with Meshtastic 
Tried 2.3.15 and 2.4.3.91d6612 Alpha (Hotfix)but had same problem. Still battery empty within 2 days in stead of 6.

**Reply by GPSFan** [reply]
I haven't had too much time to work on Meshtastic issue either, DayJob & RealLife... 
I have a Rak 4631 on a 19003 and a 12500, so I should be able to duplicate your setup, what is the capacity of your battery?
Is there anything unique about your Meshtastic configuration?
I'll load up the 2.4.3.91d6612 Alpha and see what I can measure.

**Reply by LanWolf** [reply]
Hello Ken,
I use a  [1100 mah lipo](https://www.amazon.nl/gp/product/B087LTZW61/) with this configuration, 19003,4631,12500, 0.96 oled, and a user button, [as described here with the case I use](https://www.printables.com/model/286662-rak19003-case-for-meshtastic)
